### ISSUE TITLE
Loss convergence with Horovod is poor compared to native MXNet

### ISSUE THREAD

**Issue by chandana1332** [horovod/horovod#2292]
Hello,

When comparing models trained with Horovod and native MXNet, models trained with Horovod have poorer convergence. The loss for each update when:
1. served the same batch to each GPU
2. Initialized model parameters with the same seed
have different values. The values have a small delta of ~0.01 but accumulated over a large number of updates, make a big difference. 

Is this expected behaviour? Given Native MXNet and Horovod use different implementations for all reduce, is the outcome of allreduce expected to be different?

PS: I turned off fusion to ensure consistent ( to the extent we can expect from GPUs) all reduce operations.

**Comment by karan6181** [comment]
Hey @chandana1332 , you can find more information about MXNet seed here: https://github.com/apache/incubator-mxnet/discussions/7410. It's a known issue and had a long discussion chain

**Comment by chandana1332** [comment]
@karan6181 can you elaborate on how the linked seed issue is related to the convergence issue between Horovod and MXNet? Are you saying that we will expect different results from Horovod and MXNet training because the seed is set per device?

In my experiments, I consistently see that Horovod has a poorer loss value compared to Native MXNet, tried this across 6 runs. 

I can alternatively try running both Horovod and MXNet on the same machine.

**Comment by chandana1332** [comment]
@apeforest @karan6181 any other thoughts on this ?

**Comment by karan6181** [comment]
> @karan6181 can you elaborate on how the linked seed issue is related to the convergence issue between Horovod and MXNet? Are you saying that we will expect different results from Horovod and MXNet training because the seed is set per device?

- The difference could be very minimal due to seed issue.
> 
> In my experiments, I consistently see that Horovod has a poorer loss value compared to Native MXNet, tried this across 6 runs.
- Can you please post the information such as model that u tried, training script, hyperparameters, loss values for Native MXNet and Horovod, etc. Also, if you could also attach a loss curve for Native MXNet and Horovod, then that would be great!!
> 
> I can alternatively try running both Horovod and MXNet on the same machine.
